lib/rules/no-misleading-character-class.js

```diff
@@ -193,6 +200,28 @@ const findCharacterSequences = {
 
 const kinds = Object.keys(findCharacterSequences);
 
+/**
+ * Gets the value of the given node if it's a static value other than a regular expression object,
+ * or the node's `regex` property.
+ * The purpose of this method is to provide a replacement for `getStaticValue` on Node.js 18, where `getStaticValue` returns `null` if a regular expression literal contains the `v` flag.
+ * A limitation of this method is that it can only detect a regular expression if the specified node is itself a regular expression literal node.
+ * @param {ASTNode} node The node to be inspected.
+ * @param {Scope} [initialScope] Optional scope to start finding variables. If this scope was given, this tries to resolve identifier references which are in the given node as much as possible.
+ * @returns {{ value: any } | { value: undefined, optional?: true } | { regex: { pattern: string, flags: string } } | null} The static value of the node, or `null`.
+ */
+function getStaticValueOrRegex(node, initialScope) {
+    if (node.type === "Literal" && node.regex) {
```

Comment:
This would crash if there are no arguments:

```js
/*eslint no-misleading-character-class: ["error"]*/

new RegExp();

```

```
TypeError: Cannot read properties of undefined (reading 'type')
Occurred while linting C:\projects\eslint\foo.js:3
Rule: "no-misleading-character-class"
    at getStaticValueOrRegex (C:\projects\eslint\lib\rules\no-misleading-character-class.js:213:14)
```

Reply:
My bad for not checking! Fixed in ebe11f891e218fe18e456762e68107bfbeea7f4b.